### PR TITLE
AB#8808 Sync with xmcloud-foundation-head (JSS 21.2.1)

### PR DESCRIPTION
**Pull request by chpa7**

### CODE REVIEW

File: src/rendering/package-lock.json
Comment (jflheureux) [inline src/rendering/package-lock.json:1]: The package lock file version went from `"lockfileVersion": 2` to `"lockfileVersion": 1,`. This means than an older NPM version was used to install new packages. This project should be used with nodejs 16.15.1 (as described in the 
`xmcloud.build.json` file) and the matching NPM version. Please fix this issue.

**Comment by jflheureux** [comment]
The 404 and 500 pages from the `nextjs-sxa` add-on template do not seem to be in the PR:
- https://github.com/Sitecore/jss/blob/v21.2.1/packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/404.tsx
- https://github.com/sitecorelabs/xmcloud-foundation-head/blob/main/src/sxastarter/src/pages/404.tsx

Is that intended?

File: src/rendering/scripts/scaffold-component/plugins/component.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  exec(config: ScaffoldComponentPluginConfig) {
+    const { componentName, componentPath } = config;
+    const componentFilename = `${componentName}.tsx`; // DEMO TEAM CUSTOMIZATION - Change variable name
+    const outputFilePath = path.join(this.componentRootPath, componentPath, componentFilename); // DEMO TEAM CUSTOMIZATION - Change variable name
+    const template = generateComponentSrc(componentName);
+
+    const componentOutputPath = scaffoldFile(outputFilePath, template);
+
+    // DEMO TEAM CUSTOMIZATION - Add Storybook story scaffolding
```
Comment (jflheureux) [inline src/rendering/scripts/scaffold-component/plugins/component.ts:24]: Now that scaffolding components is a plugin system, could we extract the story generation into its own plugin file?

**Comment by chpa7** [comment]
> The 404 and 500 pages from the `nextjs-sxa` add-on template do not seem to be in the PR:
> 
> * https://github.com/Sitecore/jss/blob/v21.2.1/packages/create-sitecore-jss/src/templates/nextjs-sxa/src/pages/404.tsx
> * https://github.com/sitecorelabs/xmcloud-foundation-head/blob/main/src/sxastarter/src/pages/404.tsx
> 
> Is that intended?

We already had a different, custom 404 page. Should it match the new one?

**Review by adoprog — APPROVED** [review]

**Pull request merged**